src/Send/Context/SendContext.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react'
+
+import {getDefaultAssetByNetworkId} from '../../legacy/config'
+import {YoroiWallet} from '../../yoroi-wallets'
+import {TokenId} from '../../yoroi-wallets/types'
+
+type SendProvider = {
+  wallet: YoroiWallet
+}
+
+const SendContext = React.createContext<undefined | SendContext>(undefined)
+export const SendProvider: React.FC<SendProvider> = ({children, wallet}) => {
+  const primaryTokenId = getDefaultAssetByNetworkId(wallet.networkId).identifier
+
+  const [state, dispatch] = React.useReducer(sendReducer, {
+    ...initialState,
+    selectedTokenId: primaryTokenId,
+  })
+
+  const actions = React.useRef({
+    receiverChanged: (receiver) => dispatch({type: 'receiverChanged', receiver}),
+    amountChanged: (amount) => dispatch({type: 'amountChanged', amount}),
+    sendAllChanged: () => dispatch({type: 'sendAllChanged'}),
+    tokenSelected: (selectedTokenId) => dispatch({type: 'tokenSelected', selectedTokenId}),
+    allTokensSelected: () => dispatch({type: 'allTokensSelected', primaryTokenId}),
+    resetForm: () => dispatch({type: 'resetForm', primaryTokenId}),
+  }).current
+
+  const context = React.useMemo(
+    () => ({
+      ...state,
+      ...actions,
+      selectedTokenId: state.selectedTokenId != null ? state.selectedTokenId : primaryTokenId,
```

Comment:
when resetting the state, selectedTokenId -> undefined, and that will select primaryToken

Reply:
Done